client/src/packages/Heap.ts

```diff
@@ -0,0 +1,97 @@
+/* Swap two elements of an array */
```

Comment:
Is there a reason why you need to implement the head yourself? I assume there should be some standard library which provides priorityQueue (head)?

Reply:
JS surprisingly doesn't have an in-built heap / priority queue implementation

Reply:
Add tests for your heap implementation as well in that case